Bug: Ignite CLI Panics When Retrieving Chain Info Outside Chain Working Directory

**Describe the bug**
The Ignite CLI occasionally panics with the error message `Invalid memory address or nil pointer dereference` when attempting to retrieve chain information outside the chain working directory. This issue specifically occurs during the process of obtaining chain information, leading to the unexpected behavior and panics observed in various Ignite applications.


**To reproduce**
Steps to reproduce the behavior:
1. Scaffold a chain with Ignite.
2. Serve the scaffolded chain with Ignite.
3. Install the health-monitoring app.
4. Run the health-monitoring app with the command `ignite health-monitor monitor` outside of the chain directory.

**Expected Behavior**
The ignite health-monitor monitor command should run without panicking and provide the expected monitoring output.

**Actual Behavior**
The Ignite CLI panics with the error message "invalid memory address or nil pointer dereference" when attempting to retrieve chain information outside the chain working directory.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x15fe552]

goroutine 131 [running]:
github.com/ignite/cli/v28/ignite/services/chain.(*Chain).ID(0x0?)
        /home/runner/work/cli/cli/go/src/github.com/ignite/cli/ignite/services/chain/chain.go:244 +0x12
github.com/ignite/cli/v28/ignite/services/plugin.clientAPI.GetChainInfo({{{0x3439c40?, 0x0?}}}, {0x30?, 0x28?})
        /home/runner/work/cli/cli/go/src/github.com/ignite/cli/ignite/services/plugin/client_api.go:63 +0x5c
github.com/ignite/cli/v28/ignite/services/plugin.clientAPIServer.GetChainInfo({{}, {0x3411b80?, 0xc001339830?}}, {0x3437460?, 0xc0015104e0?}, 0x412700?)
        /home/runner/work/cli/cli/go/src/github.com/ignite/cli/ignite/services/plugin/protocol.go:233 +0x31
github.com/ignite/cli/v28/ignite/services/plugin/grpc/v1._ClientAPIService_GetChainInfo_Handler({0x23a8ca0?, 0xc00084dbc0}, {0x3437460, 0xc0015104e0}, 0xc0007fb280, 0x0)
        /home/runner/work/cli/cli/go/src/github.com/ignite/cli/ignite/services/plugin/grpc/v1/service_grpc.pb.go:358 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc001514000, {0x3437460, 0xc001510450}, {0x3445720, 0xc000e7dba0}, 0xc0002e7d40, 0xc0015101e0, 0x7b1d700, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.60.1/server.go:1372 +0xe03
google.golang.org/grpc.(*Server).handleStream(0xc001514000, {0x3445720, 0xc000e7dba0}, 0xc0002e7d40)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.60.1/server.go:1783 +0xfec
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.60.1/server.go:1016 +0x59
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 94
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.60.1/server.go:1027 +0x115
```

--- a/changelog.md
+++ b/changelog.md
@@ -9,6 +9,7 @@
 ### Fixes
 
 - [#3953](https://github.com/ignite/cli/pull/3953) Fix apps `Stdout` is redirected to `Stderr`
+- [#3863](https://github.com/ignite/cli/pull/3963) Fix breaking issue for app client API when reading app chain info
 
 ## [`v28.2.0`](https://github.com/ignite/cli/releases/tag/v28.2.0)
 

--- a/ignite/cmd/plugin.go
+++ b/ignite/cmd/plugin.go
@@ -202,15 +202,14 @@ func linkPluginHook(rootCmd *cobra.Command, p *plugin.Plugin, hook *plugin.Hook)
 			}
 		}
 
-		// Get chain when the plugin runs inside an blockchain app
-		c, err := newChainWithHomeFlags(cmd)
-		if err != nil && !errors.Is(err, gomodule.ErrGoModNotFound) {
+		api, err := newAppClientAPI(cmd)
+		if err != nil {
 			return err
 		}
 
 		ctx := cmd.Context()
 		execHook := newExecutedHook(hook, cmd, args)
-		err = p.Interface.ExecuteHookPre(ctx, execHook, plugin.NewClientAPI(plugin.WithChain(c)))
+		err = p.Interface.ExecuteHookPre(ctx, execHook, api)
 		if err != nil {
 			return errors.Errorf("app %q ExecuteHookPre() error: %w", p.Path, err)
 		}
@@ -224,15 +223,14 @@ func linkPluginHook(rootCmd *cobra.Command, p *plugin.Plugin, hook *plugin.Hook)
 			err := runCmd(cmd, args)
 			// if the command has failed the `PostRun` will not execute. here we execute the cleanup step before returnning.
 			if err != nil {
-				// Get chain when the plugin runs inside an blockchain app
-				c, err := newChainWithHomeFlags(cmd)
-				if err != nil && !errors.Is(err, gomodule.ErrGoModNotFound) {
+				api, err := newAppClientAPI(cmd)
+				if err != nil {
 					return err
 				}
 
 				ctx := cmd.Context()
 				execHook := newExecutedHook(hook, cmd, args)
-				err = p.Interface.ExecuteHookCleanUp(ctx, execHook, plugin.NewClientAPI(plugin.WithChain(c)))
+				err = p.Interface.ExecuteHookCleanUp(ctx, execHook, api)
 				if err != nil {
 					cmd.Printf("app %q ExecuteHookCleanUp() error: %v", p.Path, err)
 				}
@@ -246,9 +244,8 @@ func linkPluginHook(rootCmd *cobra.Command, p *plugin.Plugin, hook *plugin.Hook)
 
 	postCmd := cmd.PostRunE
 	cmd.PostRunE = func(cmd *cobra.Command, args []string) error {
-		// Get chain when the plugin runs inside an blockchain app
-		c, err := newChainWithHomeFlags(cmd)
-		if err != nil && !errors.Is(err, gomodule.ErrGoModNotFound) {
+		api, err := newAppClientAPI(cmd)
+		if err != nil {
 			return err
 		}
 
@@ -256,7 +253,7 @@ func linkPluginHook(rootCmd *cobra.Command, p *plugin.Plugin, hook *plugin.Hook)
 		execHook := newExecutedHook(hook, cmd, args)
 
 		defer func() {
-			err := p.Interface.ExecuteHookCleanUp(ctx, execHook, plugin.NewClientAPI(plugin.WithChain(c)))
+			err := p.Interface.ExecuteHookCleanUp(ctx, execHook, api)
 			if err != nil {
 				cmd.Printf("app %q ExecuteHookCleanUp() error: %v", p.Path, err)
 			}
@@ -270,7 +267,7 @@ func linkPluginHook(rootCmd *cobra.Command, p *plugin.Plugin, hook *plugin.Hook)
 			}
 		}
 
-		err = p.Interface.ExecuteHookPost(ctx, execHook, plugin.NewClientAPI(plugin.WithChain(c)))
+		err = p.Interface.ExecuteHookPost(ctx, execHook, api)
 		if err != nil {
 			return errors.Errorf("app %q ExecuteHookPost() error : %w", p.Path, err)
 		}
@@ -333,9 +330,8 @@ func linkPluginCmd(rootCmd *cobra.Command, p *plugin.Plugin, pluginCmd *plugin.C
 		newCmd.RunE = func(cmd *cobra.Command, args []string) error {
 			ctx := cmd.Context()
 			return clictx.Do(ctx, func() error {
-				// Get chain when the plugin runs inside an blockchain app
-				c, err := newChainWithHomeFlags(cmd)
-				if err != nil && !errors.Is(err, gomodule.ErrGoModNotFound) {
+				api, err := newAppClientAPI(cmd)
+				if err != nil {
 					return err
 				}
 
@@ -348,7 +344,7 @@ func linkPluginCmd(rootCmd *cobra.Command, p *plugin.Plugin, pluginCmd *plugin.C
 					With:   p.With,
 				}
 				execCmd.ImportFlags(cmd)
-				err = p.Interface.Execute(ctx, execCmd, plugin.NewClientAPI(plugin.WithChain(c)))
+				err = p.Interface.Execute(ctx, execCmd, api)
 
 				// NOTE(tb): This pause gives enough time for go-plugin to sync the
 				// output from stdout/stderr of the plugin. Without that pause, this
@@ -694,6 +690,21 @@ func printPlugins(ctx context.Context, session *cliui.Session) error {
 		return errors.Errorf("error while printing apps: %w", err)
 	}
 	return nil
+}
+
+func newAppClientAPI(cmd *cobra.Command) (plugin.ClientAPI, error) {
+	// Get chain when the plugin runs inside an blockchain app
+	c, err := newChainWithHomeFlags(cmd)
+	if err != nil && !errors.Is(err, gomodule.ErrGoModNotFound) {
+		return nil, err
+	}
+
+	var options []plugin.APIOption
+	if c != nil {
+		options = append(options, plugin.WithChain(c))
+	}
+
+	return plugin.NewClientAPI(options...), nil
 }
 
 func flagSetPluginsGlobal() *flag.FlagSet {

